suite-native/connection-status/src/useOfflineHeaderAwareSafeAreaInsets.tsx

```diff
@@ -0,0 +1,12 @@
+import { EdgeInsets, useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import { useIsOfflineHeaderVisible } from './useIsOfflineHeaderVisible';
+
+// If offline header is visible, return 0 for top inset, otherwise return the top inset from safe area insets
+// this is beacuse the offline header is displayed on top of the screen and we don't want to add any top padding
+export const useOfflineHeaderAwareSafeAreaInsets = () => {
```

Comment:
I'm just wondering why this is needed. The `MessageSystemBannerRenderer` works the same way as the `OfflineHeader` component but a special "topAreInsetHook" is not required. 

How is it different here? I would prefer not to override the default `useSafeAreaInsets` if possible for better DX.

Reply:
What also bugs me is that now all the packages imported to the `suite-native/connection-status` (eg. settings module) are now imported to all the places where are top area insets used. That could be quite troublesome in the future. Can imagine some nasty cyclical dependency.

Reply:
this inset checking is needed otherwise if the offlineheader is visible everything with usesafeareainset has odd big offset from the top. 

Any idea how to do it differently is welcomed. 🙏

Reply:
note: out of the scope of this PR

new issue created: https://github.com/trezor/trezor-suite/issues/13659